app/forecasts/fetch.py

```diff
@@ -0,0 +1,210 @@
+        # If the previous timestamp was before noon, and the current timestamp is after noon,
+        # it means there is no noon value, and we can interpolate one.
+        result = None
+        if (self.prev_timestamp and
+                self.is_before_noon(self.prev_timestamp) and
+                self.is_after_noon(self.current_timestamp)):
+            noon = datetime.datetime(year=self.prev_timestamp.year, month=self.prev_timestamp.month,
+                                     day=self.prev_timestamp.day, hour=self.utc_noon.hour,
+                                     minute=self.utc_noon.minute, tzinfo=timezone.utc)
+            result = WeatherForecastValues(datetime=noon)
+            # x-axis is the timestamp
+            x_axis = (self.prev_timestamp.timestamp(),
+                      self.current_timestamp.timestamp())
+            # for each key value, we have a y-axis
+            y_axis = {}
+            for key, value in self.prev_values.items():
+                y_axis[key] = [value, self.current_values[key]]
+            for key, value in y_axis.items():
+                function = interp1d(x_axis, value, kind='linear')
+                interpolated_value = function(noon.timestamp())
+                # the interpolated value is in the form of an array, we just want the 1st element.
+                setattr(result, key, interpolated_value.item(0))
+        return result
+
+
+def _get_stations_for_polygon(stations, polygon: Polygon) -> List:
+    """ There could be multiple stations in this polygon, return all that match.
+    """
+    stations_in_polygon = []
+    for station in stations:
+        point = Point(station.long, station.lat)
+        logger.info('testing if %s is within %s', point, polygon)
+        if point.within(polygon):
+            stations_in_polygon.append(station)
+    return stations_in_polygon
+
+
+def _add_prediction_to_forecast(forecast: WeatherForecast, prediction: Prediction, points: List[float],
+                                noon_interpolator: NoonInterpolator):
+    forecast_values = WeatherForecastValues(
+        datetime=prediction.prediction_timestamp)
+    target_coordinate = [(forecast.station.long, forecast.station.lat)]
+    key_map = {
+        'tmp_tgl_2': 'temperature',
+        'rh_tgl_2': 'relative_humidity'
+    }
+
+    # Iterate through each of the mappings.
+    for key, target in key_map.items():
+        # Get the values.
+        values = getattr(prediction, key)
+        if values:
+            # If there are values, calculate the interpolated value, and set.
+            interpolated_value = griddata(
+                points, values, target_coordinate, method='linear')[0]
+            setattr(forecast_values, target, interpolated_value)
+            noon_interpolator.update(
+                target, interpolated_value, prediction.prediction_timestamp)
+
+    noon_forecast = noon_interpolator.calculate_noon_value()
```

Comment:
please rename this - it'll be super confusing because we also have BC FireWeather meteorologist-issued forecasts for noon-time values. They are also called "noon_forecast"